### PR TITLE
Fix - Logging in progress bar should converted to float.

### LINKED ISSUE #5481: Progress Bar on release/1.2-dev contains tensors
## Please reproduce using the BoringModel

```
class LoggerConnector:
    def __init__(self, trainer):
        self.trainer = trainer
        self._callback_metrics = MetricsHolder()
        self._evaluation_callback_metrics = MetricsHolder(to_float=True)
        self._logged_metrics = MetricsHolder()
        self._progress_bar_metrics = MetricsHolder()
```

self._logged_metrics should be `to_float=True` for progress_bar and to_float=False for merging within callbacks.

The internal API needs to be changed.

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #5481

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — CHANGES_REQUESTED** [review]
I'm pretty sure this is not the way to go :( 
1. It is not the progress bar responsibility of knowing how to convert objects to a float, especially when there is device specific logic embedded in there. 
2. We already expect trainer.progress_bar_dict to hold the values in the right format, (or lazily evalute them maybe could be an option if that is needed)
3. A user who simply wants to implement their custom progress bar will certainly not understand what this code does. 

Please reconsider this, let me know if I can help

File: pytorch_lightning/callbacks/progress.py [hunk 1]
```diff
@@ -253,6 +254,17 @@ def disable(self) -> None:
...
         self._enabled = True
 
+    def get_progress_bar_dict(self, trainer):
+        metrics_holder = trainer.logger_connector._progress_bar_metrics
+        metrics_holder._to_float = True
+        model_ref = trainer.get_model()
+        metrics_holder.convert(
+            self.trainer._device_type == DeviceType.TPU,
+            model_ref.device if model_ref is not None else model_ref
+        )
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/progress.py:264]:
```suggestion
        metrics_holder.convert(self.trainer._device_type == DeviceType.TPU, model_ref.device)
```
trainer.get_model() is never None right?

**Review by rohitgr7 — COMMENTED** [review]
just curious, if it's not converted to float in the progress bar, then what's happening here?
https://github.com/PyTorchLightning/pytorch-lightning/blob/61e1d8772d53f8543d6e2b9d7a3e60cffd975482/pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py#L253-L258

File: pytorch_lightning/callbacks/progress.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/callbacks/progress.py:264]: It is None before the first .fit() or .test() call happens :)

Comment (rohitgr7) [reply pytorch_lightning/callbacks/progress.py:264]: I meant during training.. it can't be None in the code above.

**Comment by tchaton** [comment]
> 1. It is not the progress bar responsibility of knowing how to convert objects to a float, especially when there is device specific logic embedded in there.

You are entirely. It was a temporary fix for people using this release/1.2-dev branch. I need to refactor LoggingConnector to handle things properly, but it would take me more time.

If you want to block this temporary fix, let's wait for me to properly fix it then :)

Best,
T.C

**Pull request closed without merging**